### ISSUE TITLE
Bug: If you are close to a wall while holding an object, it may phase through the wall

### ISSUE THREAD

**Issue by PiMoNFeeD** [ChaosInitiative/Portal-2-Community-Edition#1039]
### Describe the bug

https://user-images.githubusercontent.com/36343425/190915375-6f268f6c-0bf1-45ef-8975-60b3e72c526b.mp4
**Not** Jolt physics, vanilla havok-based physics
So, in the first half of the video you can see that if I push an object into the wall, it goes through it, but in the second half of the video you can see that instead of going into the wall, the object stays within the room boundaries and I get pushed into the object instead which is what should be happening I believe (the latter one happens quite rarely for me)

### Issue Map

Doesn't matter

### To Reproduce

Take an object and just try phasing it into the wall

**Comment by ozxybox** [comment]
Is this purely visual?
Is this still an issue?

**Comment by PiMoNFeeD** [comment]
I am not sure what you mean by "purely visual", if you're asking if I was able to get the cube out of the map than the answer is no, at least from my short testing
And yes, it is still an issue, it seems to not work when I look down for some reason, but it does work when I look up or I am at near 0 pitch: 

https://user-images.githubusercontent.com/36343425/217097313-fd977c02-41b8-48c6-8469-854fdd61144a.mp4

**Comment by Equalizer5118** [comment]
This looks like intended behaviour, as stock portal 2 will do the same thing. This is done for easy maneuvering through portals and to prevent wall climbing, but if anyone would like to current me feel free. The physics object does not actually disappear, its actual entity still exists in that place, but it has been made non solid to the player and its model is moved into the player.

**Comment by Luke18033** [comment]
Yeah, in addition to disabling collision with the player, Portal 2 will visually move held objects around to keep them from clipping through the camera. It probably shouldn't be moving slightly into the wall though.
